react-instantsearch-hooks/geo-search/src/Airports.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useRef } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Marker, Popup, useMapEvents } from 'react-leaflet';
+import { DivIcon } from 'leaflet';
+import { useSearchBox } from 'react-instantsearch-hooks-web';
+import { useGeoSearch } from './useGeoSearch';
+
+import type { GeoHit } from 'instantsearch.js/es/connectors/geo-search/connectGeoSearch';
+
+export function Airports() {
+  const skipViewEffect = useRef(false);
+
+  const { query, refine: refineQuery } = useSearchBox();
+  const {
+    items,
+    refine: refineItems,
+    currentRefinement,
+    clearMapRefinement,
+  } = useGeoSearch();
+
+  const onViewChange = ({ target }) => {
+    skipViewEffect.current = true;
+
+    if (query.length > 0) {
+      refineQuery('');
+    }
```

Comment:
I explained the process [here](https://github.com/algolia/doc-code-samples/pull/399/files?diff=unified&w=0#r917957294), but in a gist, moving the map will trigger a refinement on its boundaries, and having the query will filter the results on this additional parameter.

This is not ideal when searching for airports by moving the map, when a user would expect all of them to be shown instead of those matching a preset query.

Reply:
I see. Ideally the query should just be used to center the map, and then the query is not sent to Algolia anymore. That's my interpretation of what Google Maps does.